### PR TITLE
Prevent client from re-trying failed server-side requests

### PR DESCRIPTION
**Pull request by schwers**
I tested this locally by simulating an timeout in loading `/hot.json` from the frontpage. You can try this by linking `node-api-client` and then in `node-api-client/old-src/endpoints/v1.es6.js` on line 160 in `returnGETPromise` [[src]](https://github.com/reddit/node-api-client/blob/master/old-src/endpoints/v1.es6.js#L160) add the following:

```javascript
if (options.uri === 'https://www.reddit.com/hot.json') {
  err = {
    status: 503,
    message: 'fake error for testing retrying',
  };
}
```

This simulates a 503 of the listing endpoint hitting `hot.json` Note: this assumes you're logged out, you can edit the `options.uri` comparison

👓  @nramadas

### CODE REVIEW

File: assets/js/client.es6.js [hunk 1]
```diff
@@ -128,7 +129,41 @@ function initialize(bindLinks) {
     setEvents(app, hasHistAndBindLinks, render, $body);
 
     sendTimings(beginRender, props.adsEnabled);
-  });
+  };
+
+  const serverStatus = bootstrap.ctx ? bootstrap.ctx.status : 500;
+
+  if (Math.floor(serverStatus / 100) === 5) {
```
Comment (nramadas) [inline assets/js/client.es6.js]: would it be simpler to just say `((serverStatus >= 500) && (serverStatus < 600))`?

File: src/app-mixin.jsx [hunk 2]
```diff
@@ -127,15 +132,15 @@ function mixin (App) {
...
-      const title = errorMsgMap[statusCode] || errorMsgMap.default;
+      const title = ERROR_MSG_MAP[statusCode] || ERROR_MSG_MAP.default;
 
       if (!isNaN(parseInt(statusCode))) {
         ctx.status = parseInt(statusCode);
       } else {
         ctx.status = 503;
       }
 
-      Object.assign({}, ctx.props || {}, {
```
Comment (nramadas) [inline src/app-mixin.jsx]: wait, what was this even doing before?

**Comment by nramadas** [comment]
:nail_care:

File: src/app-mixin.jsx (hunk 2 above)
Comment (schwers) [inline src/app-mixin.jsx]: Exactly.

**Comment by schwers** [comment]
💇

**Comment by nramadas** [comment]
:fish:

**Pull request merged**